src/test/java/com/squareup/javapoet/ParameterSpecTest.java

```diff
@@ -48,6 +101,28 @@
     }
   }
 
+  @Test public void fieldVariableElement() {
+    TypeElement classElement = getElement(VariableElementFieldClass.class);
+    List<VariableElement> methods = fieldsIn(elements.getAllMembers(classElement));
+    VariableElement element = findFirst(methods, "name");
+
+    try {
+      ParameterSpec.get(element);
+    } catch (IllegalArgumentException exception) {
+      assertThat(exception).hasMessageThat().isEqualTo("element is not a parameter");
+    }
+  }
+
+  @Test public void parameterVariableElement() {
+    TypeElement classElement = getElement(VariableElementParameterClass.class);
+    List<ExecutableElement> methods = methodsIn(elements.getAllMembers(classElement));
+    ExecutableElement element = findFirstExecutable(methods, "foo");
+    VariableElement parameterElement = element.getParameters().get(0);
+
+    assertThat(ParameterSpec.get(parameterElement).toString())
+        .isEqualTo("@javax.annotation.Nullable java.lang.String arg0");
```

Comment:
This seems like a bug to me. This is unrelated to my changes but in `ParameterSpec`, we get the name of the element with `element.getSimpleName().toString()`. This yields `argo0` instead of `bar`. I'm not sure if this is a consequence of my test or this is the normal behavior (which seems like a bug)

Reply:
Can you repro this on master? If yes - definitely worth filing.

Reply:
Can reproduce on master. Will file a bug.